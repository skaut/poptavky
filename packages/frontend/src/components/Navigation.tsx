import styled from "@emotion/styled"
import { NavLink } from "react-router-dom"

import { theme } from "../theme"

interface NavigationItem {
  title: string
  link: string
  isActive?: boolean
}

export const Navigation = ({ items }: { items: NavigationItem[] }) => (
  <nav>
    <Container>
      {items.map((item, index) => (
        <NavigationLink key={index} to={item.link} end>
          {item.title}
        </NavigationLink>
      ))}
    </Container>
  </nav>
)

const Container = styled("div")`
  list-style: none;
  display: flex;
  margin-bottom: 24px;
  padding: 0 16px;
  border-bottom: 1px solid ${theme.colors.brand};
`

const NavigationLink = styled(NavLink)`
  display: block;
  padding: 0.2em 0.8em;
  margin-bottom: -1px;
  border: 1px solid transparent;
  border-radius: 5px 5px 0 0;

  &.active {
    border-top: 1px solid ${theme.colors.brand};
    border-right: 1px solid ${theme.colors.brand};
    border-left: 1px solid ${theme.colors.brand};
    background-color: #fff;
  }
`
