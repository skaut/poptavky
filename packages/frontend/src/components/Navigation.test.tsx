import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"

import { Navigation } from "./Navigation"

describe("Navigation component", () => {
  test("should render correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navigation
            items={[
              {
                title: "Poptávky",
                link: "/",
              },
              {
                title: "Projekty",
                link: "/projekty",
              },
            ]}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("should render correctly wyth active item", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navigation
            items={[
              {
                title: "Poptávky",
                link: "/",
              },
              {
                title: "Projekty",
                link: "/projekty",
                isActive: true,
              },
            ]}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
