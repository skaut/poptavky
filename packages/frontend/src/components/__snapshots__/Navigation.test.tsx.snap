// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should renders correctly 1`] = `
.emotion-0 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
  padding: 0 16px;
  border-bottom: 1px solid #294885;
}

.emotion-1 {
  display: block;
  padding: 0.2em 0.8em;
  margin-bottom: -1px;
  border: 1px solid transparent;
  border-radius: 5px 5px 0 0;
}

.emotion-1.active {
  border-top: 1px solid #294885;
  border-right: 1px solid #294885;
  border-left: 1px solid #294885;
  background-color: #fff;
}

<nav>
  <div
    className="emotion-0"
  >
    <a
      aria-current="page"
      className="emotion-1 active"
      href="/"
      onClick={[Function]}
    >
      Poptávky
    </a>
    <a
      className="emotion-1"
      href="/projekty"
      onClick={[Function]}
    >
      Projekty
    </a>
  </div>
</nav>
`;

exports[`Navigation component should renders correctly wyth active item 1`] = `
.emotion-0 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
  padding: 0 16px;
  border-bottom: 1px solid #294885;
}

.emotion-1 {
  display: block;
  padding: 0.2em 0.8em;
  margin-bottom: -1px;
  border: 1px solid transparent;
  border-radius: 5px 5px 0 0;
}

.emotion-1.active {
  border-top: 1px solid #294885;
  border-right: 1px solid #294885;
  border-left: 1px solid #294885;
  background-color: #fff;
}

<nav>
  <div
    className="emotion-0"
  >
    <a
      aria-current="page"
      className="emotion-1 active"
      href="/"
      onClick={[Function]}
    >
      Poptávky
    </a>
    <a
      className="emotion-1"
      href="/projekty"
      onClick={[Function]}
    >
      Projekty
    </a>
  </div>
</nav>
`;
